src/LibVLCSharp.Forms/Shared/PlaybackControls.xaml.cs

```diff
@@ -821,14 +813,113 @@ private void OnAudioTracksChanged(object sender, EventArgs e)
             UpdateAudioTracksSelectionAvailability();
         }
 
+        private void OnVideoTracksChanged(object sender, EventArgs e)
+        {
+            UpdateVideoTracksSelectionAvailability();
+        }
+
         private void OnSubtitlesTracksChanged(object sender, EventArgs e)
         {
             UpdateClosedCaptionsTracksSelectionAvailability();
         }
 
-        private async void AudioTracksSelectionButton_ClickedAsync(object sender, EventArgs e)
+
+        private void TracksButton_Clicked(object sender, EventArgs e)
+        {
+            if (ControlsPanel != null)
+                ControlsPanel.IsVisible = false;
+            var audioTrackSouces = LoadTracks(Manager.Get<AudioTracksManager>());
+            var videoTrackSouces = LoadTracks(Manager.Get<VideoTracksManager>());
+            var subtitleTracksSources = LoadTracks(Manager.Get<SubtitlesTracksManager>());
+            if (AudioTracksListView != null && VideoTracksListView != null && SubtitlesTracksListView != null)
+            {
+                AudioTracksListView.ItemsSource = audioTrackSouces;
+                VideoTracksListView.ItemsSource = videoTrackSouces;
+                SubtitlesTracksListView.ItemsSource = subtitleTracksSources;
+            }
+
+            if (TracksOverlayView != null)
+                TracksOverlayView.IsVisible = true;
+        }
+
+        private void AudioTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackViewModel)e.Item;
+            var manager = Manager.Get<AudioTracksManager>();
+            SelectTrack(manager, track, AudioTracksListView);
+        }
+
+        private void VideoTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackViewModel)e.Item;
+            var manager = Manager.Get<VideoTracksManager>();
+            SelectTrack(manager, track, VideoTracksListView);
+        }
+
+        private void SubtitlesTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackViewModel)e.Item;
+            var manager = Manager.Get<SubtitlesTracksManager>();
+            SelectTrack(manager, track, SubtitlesTracksListView);
+        }
+
+        private void SelectTrack(TracksManager manager, TrackViewModel track, ListView? tracksListview)
+        {
+            var tracks = manager.Tracks;
+            var currentTrackId = manager.CurrentTrackId;
+            if (tracks == null || track.Selected)
+                return;
+
+            var foundTrack = tracks.FirstOrDefault(t => t.Name == track.Name);
```

Comment:
Still a string comparison here. Can we be sure that the name is unique?

Reply:
I don't really know is the names come from the Media will always be unique. May be in my ViewModel, I must include the TrackId in addition to the custom Id ?